Core/Endpoints/Endpoint.php

```diff
@@ -120,11 +120,14 @@ public function getProducer()
      */
     public function consume($maxAmount = 0)
     {
-        if ($maxAmount > 0) {
-            $this->getConsumer()->setExpirationCount($maxAmount);
+        try {
+            if ($maxAmount > 0) {
+                $this->getConsumer()->setExpirationCount($maxAmount);
+            }
+            $this->getConsumer()->consume($this);
```

Comment:
Which is the reason of adding this inside a try/catch?

Reply:
I was trying to catch an exception that happens when I call the consume function. But, as long as I've been treated this inside the consume function, I think it's not necessary anymore. So I will remove the try catch statements.

Reply:
It seems the try catch has not been removed yet

Reply:
Sorry, it was removed now.